pkgs/ffigen/lib/src/config_provider/config.dart

```diff
@@ -12,368 +14,694 @@ import 'config_types.dart';
 /// headers.
 // TODO: Add a code snippet example.
 final class FfiGenerator {
-  /// Input config filename, if any.
-  final Uri? filename;
+  /// The configuration for header parsing of [FfiGenerator].
+  final Headers headers;
+
+  /// The configuration for outputting bindings.
+  final Output output;
 
   /// Path to the clang library.
+  @Deprecated('Only visible for YamlConfig plumbing.')
   final Uri? libclangDylib;
 
-  /// Output file name.
-  final Uri output;
+  /// Configuration for functions.
+  final Functions functions;
 
-  /// Output ObjC file name.
-  final Uri? outputObjC;
+  /// Configuration for structs.
+  final Structs structs;
 
-  Uri get _outputObjC => outputObjC ?? Uri.file('${output.toFilePath()}.m');
+  /// Configuration for unions.
+  final Unions unions;
 
-  /// Symbol file config.
-  final SymbolFile? symbolFile;
+  /// Configuration for enums.
+  final Enums enums;
+
+  /// Configuration for unnamed enum constants.
+  final UnnamedEnums unnamedEnums;
+
+  /// Configuration for globals.
+  final Declarations globals;
+
+  /// Configuration for macro constants.
+  final Declarations macros;
+
+  /// Configuration for typedefs.
+  final Typedefs typedefs;
+
+  /// Objective-C specific configuration.
+  ///
+  /// If `null`, will only generate for C.
+  final ObjectiveC? objectiveC;
+
+  /// Stores all the library imports specified by user including those for ffi
+  /// and pkg_ffi.
+  final List<LibraryImport> libraryImports;
+
+  /// Stores all the symbol file maps name to ImportedType mappings specified by
```

Comment:
Should probably link?

```suggestion
  /// Stores all the symbol file maps name to [ImportedType] mappings specified by
```